### PR TITLE
[train] Fix wandb/comet integration API calls

### LINKED ISSUE #38960: [train][GA] Remove remaining `checkpoint.dir_or_data` usage
https://github.com/ray-project/ray/blob/b19bcabf1e1ccaa1297f85c4bfddf68b4fed7ca1/python/ray/air/integrations/comet.py#L227

https://github.com/ray-project/ray/blob/b19bcabf1e1ccaa1297f85c4bfddf68b4fed7ca1/python/ray/air/integrations/wandb.py#L675

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Removes remaining calls to `checkpoint.dir_or_data` in the wandb/comet integrations

## Related issue number

Closes #38960

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/air/integrations/comet.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import os
 from typing import Dict, List
 
+import pyarrow.fs
+
+from ray.air.checkpoint import Checkpoint as LegacyCheckpoint
+from ray.train._checkpoint import Checkpoint as Checkpoint
```
Comment (justinvyu) [inline python/ray/air/integrations/comet.py:7]: We can use `ray.train.Checkpoint` directly, then keep the code gated w/ the env var:

```python
from ray.train import Checkpoint

...

if _use_storage_path():
    trial.checkpoint.path
else:
    trial.checkpoint.dir_or_data
```

**Review by justinvyu — COMMENTED** [review]
Thanks, this makes sense to me! Had one nit to avoid needing to update the `ray.train._checkpoint` import.

File: python/ray/air/integrations/comet.py [hunk 2]
```diff
@@ -223,19 +226,30 @@ def log_trial_save(self, trial: "Trial"):
                 name=f"checkpoint_{(str(trial))}", artifact_type="model"
             )
 
+            checkpoint_root = None
+
+            if _use_storage_context():
+                if isinstance(trial.checkpoint.storage, pyarrow.fs.LocalFileSystem):
```
Comment (justinvyu) [inline python/ray/air/integrations/comet.py:232]:
```suggestion
                if isinstance(trial.checkpoint.filesystem, pyarrow.fs.LocalFileSystem):
```

File: python/ray/air/integrations/wandb.py [hunk 3]
```diff
@@ -671,9 +672,12 @@ def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
 
     def log_trial_save(self, trial: "Trial"):
         if self.upload_checkpoints and trial.checkpoint:
-            self._trial_queues[trial].put(
-                (_QueueItem.CHECKPOINT, trial.checkpoint.dir_or_data)
-            )
+            if _use_storage_context():
+                checkpoint_root = trial.checkpoint.path
```
Comment (justinvyu) [inline python/ray/air/integrations/wandb.py:676]: Do we need to do the same local filesystem check here?

Comment (krfricke) [reply python/ray/air/integrations/wandb.py]: I think it makes sense, added. Thanks!

**Review by justinvyu — COMMENTED** [review]
Should be good after this fix:

File: python/ray/air/integrations/wandb.py [hunk 4]
```diff
@@ -671,9 +674,13 @@ def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
...
-            self._trial_queues[trial].put(
-                (_QueueItem.CHECKPOINT, trial.checkpoint.dir_or_data)
-            )
+            if _use_storage_context():
+                if isinstance(trial.checkpoint.filesystem, pyarrow.fs.LocalFileSystem):
+                    checkpoint_root = trial.checkpoint.path
+            else:
+                checkpoint_root = trial.checkpoint.dir_or_data
+
+            self._trial_queues[trial].put((_QueueItem.CHECKPOINT, checkpoint_root))
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/air/integrations/wandb.py]: Thanks for the catch!

**Review by justinvyu — APPROVED** [review]
Thanks!

**Pull request merged**